recipes/server_check_helper.rb

```diff
@@ -0,0 +1,41 @@
+require 'ipaddr'
+
+def is_server_allowed?(retry_join_list, my_ip)
+  if retry_join_list.nil? || retry_join_list.empty?()
+    return false
+  end
+
+  if retry_join_list.all? { |x| is_valid_ip_address?(x) }
+    return retry_join_list.include?(my_ip)
+  end
+
+  return true
+end
+
+def is_valid_ip_address?(s)
+  begin
+    IPAddr.new(s)
+    return true
+  rescue
+    return false
+  end
+end
+
+# This is required after changing the systemd unit file.
+def reload_systemd()
+  bash 'reload systemd daemon' do
+    user 'root'
+    code 'systemctl daemon-reload'
+  end
+end
+
+def write_metric(metric_file, metric_name, metric_help, metric_value)
+  file metric_file do
+    owner 'root'
+    mode '0644'
+    content <<-EOF.gsub(/^ */, '')
+      # HELP #{metric_name} #{metric_help}
+      #{metric_name} #{metric_value}
```

Comment:
the main concern here is that if grafana agent is not running, we will not know about this. I looked at https://roblox.atlassian.net/wiki/spaces/IC/pages/2030306932/Service+Discovery+Runbook+Decommission+A+Consul+Server and I don't see a step where we disable grafana agent, and I believe the agent is running/deployed by default in the base policy (we should confirm). Do we have a scenario in mind where we would end up with a non-working grafana agent on a rogue node ? If so, what should we consider here ?

Reply:
Oh, I see that earlier you had a POST and @sherrman suggested textfile (which was also what I originally had in mind). I'm not sure I've a great suggestion, text file are easy, POST increases the complexity (you need to handle retries, settings labels, etc).